### PR TITLE
Add socket kernel parameter host checks

### PR DESCRIPTION
**Pull request by lawrencetroup**
XRd requires socket kernel parameters relating to socket buffer sizes to be set to minimum values for production usage. Without these settings, there can be IPC loss within XRd, resulting in unpredictable behaviour. This is particularly true at higher scale.

Add host-check warnings for these values. Lower values are still permitted for lab deployments, as issues are only expected to be hit at higher scale, and intermittently.

Tests are added as part of this change to validate the new host-checks. Manual testing is also performed against hosts with valid and invalid settings.

Also update all venv dependencies to latest (and verify all tests/checks run successfully afterwards).

### CODE REVIEW

File: scripts/host-check [hunk 1]
```diff
@@ -1199,6 +1304,8 @@ BASE_CHECKS = [
     Check("Core pattern", check_core_pattern, []),
     Check("ASLR", check_userspace_aslr, []),
     Check("Linux Security Modules", check_linux_security_modules, []),
+    Check("Socket kernel parameters", check_socket_parameters, []),
+    Check("UDP kernel parameters", check_udp_parameters, []),
```
Comment (legaul-cisco) [inline scripts/host-check:1308]: Perhaps put up with the inotify checks to group kernel config checks together? (maybe the inotify checks could be one check really...)

File: scripts/host-check [hunk 2]
```diff
@@ -598,6 +598,111 @@ def check_linux_security_modules() -> CheckFuncReturn:
...
+            required_values_str += (
+                f"    net.core.{param}={required_params[param]}\n"
+            )
+            host_values_str += f"    net.core.{param}={host_params[param]}\n"
+        return (
+            CheckState.WARNING,
+            f"The kernel socket parameters are insufficient for running XRd in a\n"
+            f"production deployment. They may be used in a lab deployment, but must\n"
+            f"be increased to the required minimums for production deployment.\n"
+            f"Lower values may result at XR IPC loss and unpredictable behavior,\n"
```
Comment (legaul-cisco) [inline scripts/host-check:643]: Should this be `s/at/in/`?

Comment (lawrencetroup) [reply scripts/host-check]: Fixed (including in tests)

**Review by legaul-cisco — APPROVED** [review]

**Comment by lawrencetroup** [comment]
> Need to update changelog

Now done (including merge with previous release)

**Review by legaul-cisco — APPROVED** [review]

**Pull request merged**